engine/apps/api/permissions/__init__.py

```diff
@@ -1,5 +1,196 @@
+        INTEGRATIONS_WRITE = _generate_permission_string(Resources.INTEGRATIONS, Actions.WRITE)
+
+        ESCALATION_CHAINS_READ = _generate_permission_string(Resources.ESCALATION_CHAINS, Actions.READ)
+        ESCALATION_CHAINS_WRITE = _generate_permission_string(Resources.ESCALATION_CHAINS, Actions.WRITE)
+
+        SCHEDULES_READ = _generate_permission_string(Resources.SCHEDULES, Actions.READ)
+        SCHEDULES_WRITE = _generate_permission_string(Resources.SCHEDULES, Actions.WRITE)
+
+        CHATOPS_READ = _generate_permission_string(Resources.CHATOPS, Actions.READ)
+        CHATOPS_WRITE = _generate_permission_string(Resources.CHATOPS, Actions.WRITE)
+
+        OUTGOING_WEBHOOKS_READ = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.READ)
+        OUTGOING_WEBHOOKS_WRITE = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.WRITE)
+
+        MAINTENANCE_READ = _generate_permission_string(Resources.MAINTENANCE, Actions.READ)
+        MAINTENANCE_WRITE = _generate_permission_string(Resources.MAINTENANCE, Actions.WRITE)
+
+        API_KEYS_READ = _generate_permission_string(Resources.API_KEYS, Actions.READ)
+        API_KEYS_WRITE = _generate_permission_string(Resources.API_KEYS, Actions.WRITE)
+
+        NOTIFICATION_SETTINGS_READ = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.READ)
+        NOTIFICATION_SETTINGS_WRITE = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.WRITE)
+
+        USER_SETTINGS_READ = _generate_permission_string(Resources.USER_SETTINGS, Actions.READ)
+        USER_SETTINGS_WRITE = _generate_permission_string(Resources.USER_SETTINGS, Actions.WRITE)
+        USER_SETTINGS_ADMIN = _generate_permission_string(Resources.USER_SETTINGS, Actions.ADMIN)
+
+        OTHER_SETTINGS_READ = _generate_permission_string(Resources.OTHER_SETTINGS, Actions.READ)
+        OTHER_SETTINGS_WRITE = _generate_permission_string(Resources.OTHER_SETTINGS, Actions.WRITE)
+
+        TESTING = _generate_permission_string(Resources._IMPOSSIBLE_TESTING, Actions._IMPOSSIBLE_TESTING)
+
+    def has_permission(self, request: Request, view: ViewSetOrAPIView) -> bool:
+        action = _get_view_action(request, view)
+
+        rbac_permissions: RBACPermissionsAttribute = getattr(view, RBAC_PERMISSIONS_ATTR, None)
+
+        assert (
+            rbac_permissions is not None
+        ), f"Must define a {RBAC_PERMISSIONS_ATTR} dict on the ViewSet that is consuming the RBACPermission class"
+
+        return has_permissions(request.user.permissions, rbac_permissions.get(action, []))
+
+    def has_object_permission(self, request: Request, view: ViewSetOrAPIView, obj: typing.Any) -> bool:
+        rbac_object_permissions: RBACObjectPermissionsAttribute = getattr(view, RBAC_OBJECT_PERMISSIONS_ATTR, None)
+
+        if rbac_object_permissions:
+            action = _get_view_action(request, view)
+
+            for PermissionClass, actions in rbac_object_permissions.items():
```

Comment:
Probably that's subjective, but I'd prefer to change `PermissionClass` to `permission_class`. When scanning for the first time I thought that `PermissionClass` is something imported from other file.